2019Labs/CustomSecurityContent/documentation/lab3_oval.adoc

```diff
@@ -0,0 +1,487 @@
+
+When the build finishes, let's take a look at the RHEL8 OSPP profile:
+
+----
+firefox build/guides/ssg-rhel8-guide-ospp.html
+----
+
+In this track, we will focus on the `accounts_tmout` rule.
+In order to find the rule entry it in the guide, press `Ctrl+F` and search for the `Set Interactive Session Timeout` string, which is the rule title.
+
+The description says:
+
+____
+
+Setting the TMOUT option in /etc/profile ensures that all user sessions will terminate based on inactivity. The TMOUT setting in /etc/profile should read as follows:
+
+TMOUT=600
+____
+
+
+When dealing with the rule check, we will focus on those two aspects:
+
+- What if the timeout is set to 100?
+Is it more or less secure?
++
+Having shorter time interval between inactivity and logout is more bothersome for the user, but it is a stricter requirement.
+Therefore, we would like to make sure that the if a rule that requires `TMOUT=600` also allows `TMOUT=100`.
+
+- The rule description of having the `TMOUT=...` statement in a config file is accurate, but guides on the Internet often recommend to have `export TMOUT=...` there.
+We want to make sure that the rule's check will allow both forms - with and without `export`, although the `export` keyword is not required.
+
+
+==== Bash remediation
+
+Let's examine the bash remediation - as you know from the introductory lab, you can either
+- scroll down the rule entry in the HTML guide and click the `(show)` link next to the `Remediation Shell Script` label, or
+- open the `linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh` file.
+
+....
+
+if grep --silent ^TMOUT /etc/profile ; then
+        sed -i "s/^TMOUT.*/TMOUT=$var_accounts_tmout/g" /etc/profile
+else
+        echo -e "\n# Set TMOUT to $var_accounts_tmout per security requirements" >> /etc/profile
+        echo "TMOUT=$var_accounts_tmout" >> /etc/profile
+fi
+
+....
+
+We can see that the remediation is in sync with the description - it either adds the `TMOUT` line, or it modifies the `TMOUT` assignment if one already exists.
+However, it doesn't address invalid assignments in the `/etc/profile.d` directory.
+
+
+==== OVAL check
+
+Let's move on to the OVAL check in `linux_os/guide/system/accounts/accounts-session/accounts_tmout/oval/shared.xml`.
+That file is much more complicated, so let's examine it piecewise:
+
+We have the leading `definition` element:
+
+....
+
+  <definition class="compliance" id="accounts_tmout" version="2">
+    <metadata>
+      <title>Set Interactive Session Timeout</title>
+      <affected family="unix">
+        <platform>multi_platform_rhel</platform>
+        <platform>multi_platform_fedora</platform>
+        <platform>multi_platform_ol</platform>
+      </affected>
+      <description>Checks interactive shell timeout</description>
+    </metadata>
+    <criteria operator="OR">
+      <criterion comment="TMOUT value in /etc/profile >= var_accounts_tmout" test_ref="test_etc_profile_tmout" />
```

Comment:
Does `>` get automatically escaped?

Reply:
Surprisingly, this is OK: https://stackoverflow.com/questions/1091945/what-characters-do-i-need-to-escape-in-xml-documents

Reply:
OK